### PR TITLE
CAPBM increased coverage for machine manager unit test

### PR DESCRIPTION
**Pull request by Jaakko-Os**

### CODE REVIEW

File: baremetal/machine_manager_test.go [hunk 1]
```diff
@@ -25,11 +41,32 @@ const (
...
+		Name:      "mybmmachine",
+		Namespace: "myns",
+	},
+	Image: capbm.Image{
+		URL:      testImageURL,
+		Checksum: testImageChecksumURL,
+	},
+	HostSelector: capbm.HostSelector{},
+}
+
```
Comment (kashifest) [inline baremetal/machine_manager_test.go]: I am wondering if declaring these different bmmspecs would make more sense if it is done in testcases.

File: baremetal/machine_manager_test.go [hunk 2]
```diff
@@ -51,13 +88,28 @@ var consumerRefSome = &corev1.ObjectReference{
...
+var expectedImg = &bmh.Image{
+	URL:      testImageURL,
+	Checksum: testImageChecksumURL,
+}
+
+var expectedImgTest = &bmh.Image{
+	URL:      testImageURL + "test",
+	Checksum: testImageChecksumURL + "test",
+}
+
```
Comment (kashifest) [inline baremetal/machine_manager_test.go]: Same comment as previous one

File: baremetal/machine_manager_test.go [hunk 3]
```diff
@@ -51,13 +88,28 @@ var consumerRefSome = &corev1.ObjectReference{
...
 }
 
+var bmhSpecTestImg = &bmh.BareMetalHostSpec{
+	ConsumerRef: consumerRef,
+	Image:       expectedImgTest,
+}
+
 var bmhSpecSomeImg = &bmh.BareMetalHostSpec{
 	ConsumerRef: consumerRefSome,
 	Image: &bmh.Image{
```
Comment (kashifest) [inline baremetal/machine_manager_test.go]: Same comment as previous one

File: baremetal/machine_manager_test.go (hunk 1 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager_test.go]: The structs defined here are mainly used in two or more test cases, avoiding repetition.
I would prefer to have it like this.

File: baremetal/machine_manager_test.go (hunk 2 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager_test.go]: I would prefer to have it like this.

File: baremetal/machine_manager_test.go (hunk 3 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager_test.go]: I would prefer to have it like this.

**Pull request merged**